src/flinkSql/multiSocketManager.ts

```diff
@@ -0,0 +1,217 @@
+import { TextDocument } from "vscode";
+import { Message } from "vscode-languageclient/node";
+import { CCLOUD_CONNECTION_ID } from "../constants";
+import { Logger } from "../logging";
+import { SIDECAR_PORT } from "../sidecar/constants";
+import { ConnectionStatus, LanguageServerConnection } from "./socketConnectionManager";
+
+const logger = new Logger("flinkSql.multiServerManager");
+
+interface ServerInfo {
+  url: string;
+  manager: LanguageServerConnection;
+  documentFilter: (document: TextDocument) => boolean;
+}
+
+/**
+ *  Manages multiple websocket server connections, routes messages based on document filters
+ */
+export class MultiServerManager {
```

Comment:
nit: maybe make this match the name of the file or vice versa, had to do a double-take reading it